src/dmini.c

```diff
@@ -0,0 +1,921 @@
+    if (!buffer)
+    {
+        return DMINI_ERR_MEMORY;
+    }
+    
+    dmini_section_t* current_section = ctx->sections; // Start with global section
+    
+    char* line = buffer;
+    char* next_line;
+    
+    while (line && *line)
+    {
+        // Find end of line
+        next_line = line;
+        while (*next_line && *next_line != '\n' && *next_line != '\r')
+        {
+            next_line++;
+        }
+        
+        // Null terminate current line
+        if (*next_line)
+        {
+            *next_line = '\0';
+            next_line++;
+            // Skip \r\n combinations
+            if (*next_line == '\n' || *next_line == '\r')
+            {
+                next_line++;
+            }
+        }
+        
+        // Trim whitespace
+        line = trim_whitespace(line);
+        
+        // Skip empty lines and comments
+        if (*line == '\0' || *line == ';' || *line == '#')
+        {
+            line = next_line;
+            continue;
+        }
+        
+        // Check for section header
+        if (*line == '[')
+        {
+            char* section_end = line + 1;
+            while (*section_end && *section_end != ']')
+            {
+                section_end++;
+            }
+            
+            if (*section_end == ']')
+            {
+                *section_end = '\0';
+                char* section_name = trim_whitespace(line + 1);
+                
+                current_section = get_or_create_section(ctx, section_name);
+                if (!current_section)
+                {
+                    Dmod_Free(buffer);
+                    return DMINI_ERR_MEMORY;
+                }
+            }
+            
+            line = next_line;
+            continue;
+        }
+        
+        // Parse key=value
+        char* equals = line;
+        while (*equals && *equals != '=')
+        {
+            equals++;
+        }
+        
+        if (*equals == '=')
+        {
+            *equals = '\0';
+            char* key = trim_whitespace(line);
+            char* value = trim_whitespace(equals + 1);
+            
+            if (*key)
+            {
+                int result = set_pair_in_section(current_section, key, value);
+                if (result != DMINI_OK)
+                {
+                    Dmod_Free(buffer);
+                    return result;
+                }
+            }
+        }
+        
+        line = next_line;
+    }
+    
+    Dmod_Free(buffer);
+    return DMINI_OK;
+}
+
+int dmini_parse_file(dmini_context_t ctx, const char* filename)
+{
+    if (!ctx || !filename)
+    {
+        return DMINI_ERR_INVALID;
+    }
+    
+    // Open file using SAL
+    void* file = Dmod_FileOpen(filename, "r");
+    if (!file)
+    {
+        return DMINI_ERR_FILE;
+    }
+    
+    // Get file size
+    size_t file_size = Dmod_FileSize(file);
+    
+    // Allocate buffer
+    char* buffer = (char*)Dmod_Malloc(file_size + 1);
```

Comment:
Nie czytaj całego pliku na raz. To ma być używane w embedded i szkoda pamięci - czytaj lepiej linia po linii